lib/pulsar/producer.rb

```diff
@@ -22,9 +22,9 @@
 module Pulsar
   class Producer
     module RubySideTweaks
-      def send(message)
+      def send(message, partitionKey='')
         unless message.is_a?(Pulsar::Message)
-          message = Pulsar::Message.new(message)
+          message = Pulsar::Message.new(message, partitionKey)
```

Comment:
I'm slightly concerned about unexpected behavior if we have e.g.

```
producer = ...
message = Pulsar::Message.new("message contents")
producer.send(message, "my-partition")
```
which may not be how anyone's currently using, but is allowed

would it be too onerous to leave the partition key handling out of `Producer#send` and instead just have users that want a partition key on their messages make the `Message` explicitly first and then pass that to producer.send instead of the raw data?

Reply:
We would still need the rest of this PR to allow putting the partition key on the `Message`

Reply:
yep, this would be very good

Reply:
in this case, I am going to make partition_key an optional value on the C++ constructor, as well as allowing it to be explicitly set